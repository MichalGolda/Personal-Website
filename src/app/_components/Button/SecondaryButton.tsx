import Button from "./Button";
import type { ButtonProps } from "./Button";

const SecondaryButton: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  className = "",
  children,
  ...props
}) => {
  return (
    <Button
      className={`bg-secondary text-white hover:bg-secondary/90 active:bg-secondary/80 border border-transparent ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};

export default SecondaryButton;
