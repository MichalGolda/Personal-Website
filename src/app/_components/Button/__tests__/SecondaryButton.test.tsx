import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import SecondaryButton from "../SecondaryButton"

describe("SecondaryButton component", () => {
    it("renders without errors", () => {
        render(<SecondaryButton type="button">Button test</SecondaryButton>)
    
        const button = screen.getByRole("button")

        expect(button).toBeInTheDocument()
        expect(button).toHaveAttribute("type", "button")
        expect(button).toHaveClass("bg-secondary", "text-white", "hover:bg-secondary/90", "active:bg-secondary/80", "border-transparent")
    })

    it("renders unchanged", () => {
        const { container } = render(<SecondaryButton type="button">Button test</SecondaryButton>)

        expect(container).toMatchSnapshot()
    })
})