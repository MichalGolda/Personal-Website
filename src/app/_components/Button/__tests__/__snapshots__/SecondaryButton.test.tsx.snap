// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton component renders unchanged 1`] = `
<div>
  <button
    class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-secondary text-white hover:bg-secondary/90 active:bg-secondary/80 border border-transparent "
    type="button"
  >
    Button test
  </button>
</div>
`;
